project-koku/koku
# Address sources sentry errors

There are several sources sentry errors that this PR is aiming to address:
* CommitFailedError, Heartbeat session expired, Offset commit failed (Moving to synchronous kafka library)
* Unknown source type errors (source_type = `""`).  Also addresses the empty source db entries we have seen in CI. (Better handling sources-api connection errors)

**Testing**
Local platform sources smoke tests passed.

1. Create a source and bring database down while processing.  Verify source creation is retried and successful when db comes back.
2. Stop sources-api while processing a message.  Verify message is re-tried and successful when sources-api comes back online.
3. Restart kafka and verify source creation is successful.
4. Shutdown sources-client while in the middle of processing.  Verify message is re-processed and source creation is successful when pod restarts.

**Test Results**
[sources_sync_ut.txt](https://github.com/project-koku/koku/files/4683894/sources_sync_ut.txt)
